### PR TITLE
[MM-9216] Add date separator per search result item

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Add date separator per search result item

#### Ticket Link
Jira ticket: [MM-9216](https://mattermost.atlassian.net/browse/MM-9216)

#### Checklist
- [x] Has UI changes

#### Device Information
This PR was tested on: [Simulator iPhone 8, 11.2] 

#### Screenshots
Posted a confirmation at UX Design Channel - https://pre-release.mattermost.com/core/pl/4aukagrynfddugu3sjio6731zc (will update this PR depending on the design decision on that post)

screen shot 2018-04-13 at 4 22 50 am

### CODE REVIEW

File: app/selectors/post_list.js [hunk 1]
```diff
@@ -80,3 +80,26 @@ export function makePreparePostIdsForPostList() {
...
+        (posts) => {
+            const out = [];
+
+            // Iterating through the posts from oldest to newest
+            for (let i = posts.length - 1; i >= 0; i--) {
+                const post = posts[i];
+                const postDate = new Date(post.create_at);
+
+                out.push(post.id);
+                out.push(DATE_LINE + postDate.toString());
```
Comment (enahum) [inline app/selectors/post_list.js]: If you have more than one result on the same date you’ll get more than one date separator

**Comment by saturninoabril** [comment]
Awesome! I'll use your implementation. Will just wait for that to get merge, then will update this PR.
Thanks!

**Comment by saturninoabril** [comment]
Reopened this PR to maintain each post with date separator for recent, flagged and search posts.

**Comment by saturninoabril** [comment]
This is now ready for review.

**Pull request merged**